### PR TITLE
Mindlamp logic update

### PR DESCRIPTION
**Pull request by kcho**
Mindlamp API doesn't have checksum functions. Thus, to compare local data to the data on mindlamp server, the data on the mindlamp server must be downloaded first to compute checksum locally. This step is required to sync data between the data aggregation server and mindlamp server. However, since mindlamp team highlighted that the data on mindlamp servers don't change once stored on the server, it could be safely assumed that the data stored more than 2 days ago should not change.

This PR makes lochness.mindlamp module to use different logics for data that has been more than 2 days and data within 2 days (from the current date).

- Within 2 days from the current date
  - Download the data from mindlamp server and compare checksum before overwriting existing data
  - Once downloaded, save checksum in a hidden checksum file

- After 2 days from the current date
  - Skip downloading data from mindlamp server if the hidden checksum file exists for the data.

However, if there is no data file, lochness.mindlamp will run mindlamp API to download the missing data over the days configured in the external lochness configuration file.

### CODE REVIEW

File: lochness/mindlamp/__init__.py [hunk 1]
```diff
@@ -183,26 +194,28 @@ def sync(Lochness: 'lochness.config',
                         dst_folder,
                         f'{subject_id}_{subject.study}_{data_name}_'
                         f'{date_str}_sound.mp3')
-                if is_transferred_and_removed(Lochness, sound_dst):
-                    continue
+                # if is_transferred_and_removed(Lochness, sound_dst):
+                    # continue
```
Comment (tashrifbillah) [inline lochness/mindlamp/__init__.py:198]: There is a problem with Kevin's block commenting. Block comments should start with all `#` in the same column.

File: lochness/mindlamp/__init__.py [hunk 2]
```diff
@@ -152,26 +153,36 @@ def sync(Lochness: 'lochness.config',
...
+                    logger.debug(f'{data_name} data has been downloaded for '
+                                 f'{date_str} - skip downloading')
                     continue
 
-                prev_file_sha256 = get_sha(dst)
-            else:
-                prev_file_sha256 = ''
+            elif days_from_ct < 2 and Path(dst).is_file():
+                # potentially within 24 hours from the data acquisition, thus
+                # check if there is any changes in the data on the source
```
Comment (tashrifbillah) [inline lochness/mindlamp/__init__.py:172]: Suggestion:

                # potentially within 24 hours from the data acquisition, data may change
                # so check if there is any changes in the data on the source

**Comment by tashrifbillah** [comment]
This PR does not have any work for the trailing comment:

> However, if there is no data file, lochness.mindlamp will run mindlamp API to download the missing data over the days configured in the external lochness configuration file.

I suggest removing it from description and/or putting it separately as a comment.

**Review by tashrifbillah — APPROVED** [review]
Tashrif does not have any code related comments. His comments are minor, intended for clarity and record purposes.

File: lochness/utils/source_check.py [hunk 3]
```diff
@@ -274,7 +274,7 @@ def collect_box_files_info(keyring) -> pd.DataFrame:
     def collect_out(df):
         results.append(df)
 
-    pool = Pool(10)
+    pool = Pool(2)
```
[the remark this answers is not in the document]
Comment (kcho) [reply lochness/utils/source_check.py:277]: The current pronet lochness server is not strong enough to accomodate 10 threads parallel jobs.

File: lochness/mindlamp/__init__.py (hunk 1 above)
Comment (kcho) [reply lochness/mindlamp/__init__.py:198]: 4c4d7d0d39ee409deaae44492298781a8c728c7c

File: lochness/mindlamp/__init__.py (hunk 2 above)
Comment (kcho) [reply lochness/mindlamp/__init__.py]: 0ebf9f49c5fdc035c8dd4028341049a5f9c1931c

**Pull request merged**